### PR TITLE
Cache config list to increase performance populating remote branches

### PR DESCRIPTION
**Pull request by ccallebs**
This is a stab at addressing #291. I may be off the mark, but locally this seems to work fine.

### CODE REVIEW

**Comment by perlun** [comment]
@ccallebs Thanks for the effort. If possible, please rebase this towards master so we are CI green, and we'll try to get it reviewed asap.

**Comment by ccallebs** [comment]
@perlun Rebased and ready for review.

File: lib/git/lib.rb [hunk 1]
```diff
@@ -457,10 +457,12 @@ def config_list
         parse_config_list command_lines('config', ['--list'])
       end
 
-      if @git_dir
-        Dir.chdir(@git_dir, &build_list)
-      else
-        build_list.call
+      @config_list ||= begin
+        if @git_dir
+          Dir.chdir(@git_dir, &build_list)
```
Comment (perlun) [inline lib/git/lib.rb:462]: What if `@git_dir` changes after the content is cached? Can it happen?

Comment (perlun) [reply lib/git/lib.rb:462]: Ping @ccallebs

**Comment by perlun** [comment]
@ccallebs See my feedback above.

**Comment by jcouball** [comment]
Closing this for lack of author feedback.  Please reopen and provide feedback for further consideration.

**Pull request closed without merging**